src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Formatters/StandardClassLibraryFormatter.cs

```diff
@@ -108,13 +108,76 @@ private DecimalFormatter()
 
         public void Serialize(ref MessagePackWriter writer, decimal value, MessagePackSerializerOptions options)
         {
-            writer.Write(value.ToString(CultureInfo.InvariantCulture));
-            return;
+            var dest = writer.GetSpan(MessagePackRange.MaxFixStringLength);
+            if (System.Buffers.Text.Utf8Formatter.TryFormat(value, dest.Slice(1), out var written))
+            {
+                // write string header
+                dest[0] = (byte)(MessagePackCode.MinFixStr | written);
+                writer.Advance(written + 1);
+            }
+            else
+            {
+                writer.Write(value.ToString(CultureInfo.InvariantCulture));
+            }
         }
 
         public decimal Deserialize(ref MessagePackReader reader, MessagePackSerializerOptions options)
         {
-            return decimal.Parse(reader.ReadString(), CultureInfo.InvariantCulture);
+            if (!(reader.ReadStringSequence() is ReadOnlySequence<byte> sequence))
+            {
+                throw new MessagePackSerializationException(string.Format("Unexpected msgpack code {0} ({1}) encountered.", MessagePackCode.Nil, MessagePackCode.ToFormatName(MessagePackCode.Nil)));
+            }
+
+            if (sequence.IsSingleSegment)
+            {
+                if (System.Buffers.Text.Utf8Parser.TryParse(sequence.First.Span, out decimal result, out _))
+                {
+                    return result;
+                }
+            }
+            else
+            {
+                if (sequence.Length < 128)
+                {
+                    Span<byte> span = stackalloc byte[(int)sequence.Length];
+                    sequence.CopyTo(span);
+                    if (System.Buffers.Text.Utf8Parser.TryParse(span, out decimal result, out _))
+                    {
+                        return result;
+                    }
+                }
+                else
+                {
+                    var rentArray = ArrayPool<byte>.Shared.Rent((int)sequence.Length);
+                    try
+                    {
+                        sequence.CopyTo(rentArray);
+                        if (System.Buffers.Text.Utf8Parser.TryParse(rentArray.AsSpan(0, (int)sequence.Length), out decimal result, out _))
+                        {
+                            return result;
+                        }
+                    }
+                    finally
+                    {
+                        ArrayPool<byte>.Shared.Return(rentArray);
+                    }
+                }
+            }
+
+            // fallback
```

Comment:
Under what condition do you need this fallback? When `TryParse` fails? Why not just throw instead of allocating more memory to try again only for it to fail? Is it that you want a better exception that describes why parse failed?

Reply:
I've used same manner of TryFormat.
However, it is true that TryParse certainly means failure.
Fixed to throw exception.